lib/hccrawler.js

```diff
@@ -459,72 +443,62 @@ class HCCrawler extends EventEmitter {
    * @return {!Promise<!boolean>}
    * @private
    */
-  _checkRequested(options) {
-    if (!options.skipDuplicates) return Promise.resolve(false);
+  async _checkRequested(options) {
+    if (!options.skipDuplicates) return false;
     const key = generateKey(options);
-    return this._cache.get(key).then(value => !!value);
-  }
-
-  /**
-   * @param {!Object} options
-   * @return {!Promise<!boolean>}
-   * @private
-   */
-  _markRequested(options) {
-    if (!options.skipDuplicates) return Promise.resolve(false);
-    const key = generateKey(options);
-    return this._cache.set(key, '1');
+    const value = await this._cache.get(key);
+    return !!value;
   }
 
   /**
    * @param {!Object} options
    * @return {!Promise}
    * @private
    */
-  _removeExists(options) {
```

Comment:
it doesn't have to remove `exists` mark because retry doesn't go though `_checkRequested` method.